pkg/utils/resources/suite_test.go

```diff
@@ -0,0 +1,725 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"testing"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/samber/lo"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	"sigs.k8s.io/karpenter/pkg/test"
+)
+
+// ExpectResources expects all the resources in expected to exist in real with the same values
+func ExpectResources(expected, real v1.ResourceList) {
+	GinkgoHelper()
+	for k, v := range expected {
+		realV := real[k]
+		Expect(v.Value()).To(BeNumerically("~", realV.Value()))
+	}
+}
+
+func TestResources(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Resources Suite")
+}
+
+var _ = Describe("Resources Handling", func() {
```

Comment:
Looks like we still need to add some conditional skips in these tests if the k8s minor version is less than 29.

Reply:
Nevermind, forgot that these tests don't actually ever hit an API server, this is good as is.